pkg/securitycontext/util.go

```diff
@@ -66,6 +66,18 @@ func DetermineEffectiveSecurityContext(pod *v1.Pod, container *v1.Container) *v1
 		*effectiveSc.SELinuxOptions = *containerSc.SELinuxOptions
 	}
 
+	if containerSc.WindowsOptions != nil {
+		// only override fields that are set at the container level, not the whole thing
+		if effectiveSc.WindowsOptions == nil {
+			effectiveSc.WindowsOptions = &v1.WindowsSecurityContextOptions{}
+		}
+		if containerSc.WindowsOptions.GmsaCredentialSpecName != nil || containerSc.WindowsOptions.GmsaCredentialSpec != nil {
+			// both GMSA fields go hand in hand
+			*effectiveSc.WindowsOptions.GmsaCredentialSpecName = *containerSc.WindowsOptions.GmsaCredentialSpecName
```

Comment:
won't `*effectiveSc.WindowsOptions.GmsaCredentialSpecName` and `*effectiveSc.WindowsOptions.GmsaCredentialSpecName` NPE?

Reply:
one of `*containerSc.WindowsOptions.GmsaCredentialSpecName` and `*containerSc.WindowsOptions.GmsaCredentialSpec` could be nil here and NPE

would this be better?
```go
if containerSc.WindowsOptions.GmsaCredentialSpecName != nil || containerSc.WindowsOptions.GmsaCredentialSpec != nil {
  if containerSc.WindowsOptions.GmsaCredentialSpecName == nil {
    effectiveSc.WindowsOptions.GmsaCredentialSpecName = nil
  } else {
    s := *containerSc.WindowsOptions.GmsaCredentialSpecName
    effectiveSc.WindowsOptions.GmsaCredentialSpecName = &s
  }

  if containerSc.WindowsOptions.GmsaCredentialSpec == nil {
    effectiveSc.WindowsOptions.GmsaCredentialSpec = nil
  } else {
    s := *containerSc.WindowsOptions.GmsaCredentialSpec
    effectiveSc.WindowsOptions.GmsaCredentialSpec = &s
  }
}
```